### PR TITLE
Crypto part 3

### PR DESCRIPTION
**Pull request by arthurchan35**
Hey Brian, /cc @arthurchan35

here we are extracting the different providers to their own modules and making the CryptoHashGeneratorImpl accept the provider as a dependency, so different algorithms can be supported.

Let us know if the structure is ok so far.

Carlos.

### CODE REVIEW

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/java/com/liferay/portal/crypto/hash/provider/bcrypt/internal/BCryptCryptoHashProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/java/com/liferay/portal/crypto/hash/provider/bcrypt/internal/BCryptCryptoHashProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/MessageDigestCryptoHashProvider.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
```
Comment (liferay-appsec) [inline modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-message-digest/src/main/java/com/liferay/portal/crypto/hash/provider/message/digest/internal/MessageDigestCryptoHashProvider.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by arthurchan35** [comment]
ci:forward

**Pull request closed without merging**